packages/keyring-api/src/api/caip.ts

```diff
@@ -1,16 +1,28 @@
 import { definePattern } from '@metamask/keyring-utils';
 import { type Infer } from '@metamask/superstruct';
-import type { CaipAssetId, CaipAssetType } from '@metamask/utils';
+import type { CaipChainId, CaipAssetId, CaipAssetType } from '@metamask/utils';
 import {
+  CAIP_CHAIN_ID_REGEX,
   CAIP_ASSET_TYPE_REGEX,
   CAIP_ASSET_ID_REGEX,
+  isCaipChainId,
   isCaipAssetType,
   isCaipAssetId,
 } from '@metamask/utils';
 
 const CAIP_ASSET_TYPE_OR_ID_REGEX =
   /^(?<chainId>(?<namespace>[-a-z0-9]{3,8}):(?<reference>[-_a-zA-Z0-9]{1,32}))\/(?<assetNamespace>[-a-z0-9]{3,8}):(?<assetReference>[-.%a-zA-Z0-9]{1,128})(\/(?<tokenId>[-.%a-zA-Z0-9]{1,78}))?$/u;
 
+/**
+ * A CAIP-2 chain ID, i.e., a human-readable agnostic chain ID.
+ */
+export const CaipChainIdStruct = definePattern<CaipChainId>(
```

Comment:
This already exists in `utils`, do we need to redefine it?

Reply:
Not really, the only reason why we re-define those for now is because our `definePattern` gives better error messages, like here: https://github.com/MetaMask/accounts/blob/v23.0.0/packages/keyring-snap-sdk/src/rpc-handler.test.ts#L604

IMO, that would be great that we move those to `utils` at some point.

But type-wise, they truly are the same, yes.

Reply:
See:
- https://github.com/MetaMask/utils/pull/228